web/lumpTypeView.ts

```diff
@@ -572,3 +578,169 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
         }
     });
 }
+
+export function LumpTypeViewColormapAll(scaleX:number = 2, scaleY:number = 2) {
+    return new LumpTypeView({
+        name: "Colormap (all)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            const cmap = lumps.WADColorMap.from(lump);
+
+            // Get PLAYPAL, or use default if it isn't found.
+            const files = getWadFileList(lump);
+            const playpal = files.getPlaypal();
+
+            // Set up canvas
+            const displayWidth = playpal.getColorPerPaletteCount() * scaleX;
+            const displayHeight = cmap.getMapCount() * scaleY;
+            const canvas = util.createElement({
+                tag: "canvas",
+                class: "lump-view-image",
+                appendTo: root
+            });
+            canvas.width = displayWidth;
+            canvas.height = displayHeight;
+
+            const context = canvas.getContext("2d") as CanvasRenderingContext2D;
+            const data = cmap.getPixelDataRGBA(playpal, 0);
+
+            // Draw the colormap
+            for (let cmapIdx = 0; cmapIdx < cmap.getMapCount(); cmapIdx++) {
+                for (let palIdx = 0; palIdx < playpal.getColorPerPaletteCount(); palIdx++) {
+
+                    // Figure out where to paint
+                    const row = cmapIdx * scaleY;
+                    const col = palIdx * scaleX;
+                    const width = scaleX;
+                    const height = scaleY;
+
+                    // Data index
+                    const didx = (palIdx + cmapIdx * playpal.getColorPerPaletteCount()) * 4;
+
+                    // Get RGB string
+                    let rgb = data.readUInt32BE(didx).toString(16);
+                    while (rgb.length < 8) {
```

Comment:
Use the `String.padStart` method instead of a while loop. For example, L621 could read:

``` ts
const rgb: string = data.readUInt32BE(didx).toString(16).padStart(8, "0");
```